src/seedu/addressbook/commands/FindByTagCommand.java

```diff
@@ -0,0 +1,68 @@
+package seedu.addressbook.commands;
+
+import java.util.*;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.tag.Tag;
+
+/**
+ * Finds and lists all persons in address book whose name contains any of the argument keywords.
+ * Keyword matching is case sensitive.
+ */
+public class FindByTagCommand extends Command {
+
+    public static final String COMMAND_WORD = "findByTag";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all persons whose tags match any of "
+            + "the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
+            + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
+            + "Example: " + COMMAND_WORD + " school work CCA";
+
+    private final Set<Tag> tags;
+
+    public FindByTagCommand(Set<String> tags) {
+        final Set<Tag> tagSet = new HashSet<>();
+        for (String tagName : tags) {
+            try {
+                Tag newTag = new Tag(tagName);
+                if (!tagSet.contains(newTag)) {
+                    tagSet.add(newTag);
+                }
+            } catch (IllegalValueException ive) {}
```

Comment:
do remember to handle the exception. If its intentionally to ignore, can add a comment to explain why is that so.